src/tensor_ops/gelu/gelu.cu

```diff
@@ -1,4 +1,6 @@
 #include "unary_op_macros.cuh"
+#define _USE_MATH_DEFINES
```

Comment:
What does this do exactly? Also have you tried compiling all this on a non GNU and MSVC device?

Reply:
From the [MSVC documentation](https://learn.microsoft.com/en-us/cpp/c-runtime-library/math-constants?view=msvc-170)
> The math constants aren't defined in Standard C/C++. To use them, you must first define `_USE_MATH_DEFINES`, and then include `<cmath>` or `<math.h>`.

The `#define` is used here to essentially set a flag so that the math constants are defined in the `math.h` header file. 
If we take a look at the ([relevant documentation](https://learn.microsoft.com/en-us/cpp/preprocessor/hash-define-directive-c-cpp?source=recommendations&view=msvc-170)) we can see how it works:

> A `#define` without a token-string removes occurrences of identifier from the source file. The identifier remains defined and **can be tested** by using the `#if defined` and `#ifdef` directives.

(Emphasis mine)

---

I have not tested with other toolchains than `stable-x86_64-pc-windows-gnu` and `stable-x86_64-pc-windows-msvc`.